src/main/java/at/ac/tuwien/kr/alpha/grounder/transformation/aggregates/encoders/MinMaxEncoder.java

```diff
@@ -0,0 +1,107 @@
+import at.ac.tuwien.kr.alpha.common.program.InputProgram;
+import at.ac.tuwien.kr.alpha.common.terms.Term;
+import at.ac.tuwien.kr.alpha.grounder.parser.ProgramParser;
+import at.ac.tuwien.kr.alpha.grounder.transformation.aggregates.AggregateRewritingContext;
+import at.ac.tuwien.kr.alpha.grounder.transformation.aggregates.AggregateRewritingContext.AggregateInfo;
+
+public class MinMaxEncoder extends AbstractAggregateEncoder {
+
+	//@formatter:off
+	private static final String MINMAX_ELEMENT_ORDERING =
+			"$id$_element_tuple_less_than(ARGS, LESS, THAN) :- " +
+			"	$id$_element_tuple(ARGS, LESS), $id$_element_tuple(ARGS, THAN), LESS < THAN.";
+	
+	/**
+	 * If the literal to encode binds a variable (e.g. "X = #min{...}", "Y = #max{...}"), use this template for the result rule of the aggregate encoding.
+	 */
+	private static final ST BINDING_LITERAL_RESULT_RULE = Util.aspStringTemplate(
+			"$aggregate_result$(ARGS, M) :- $id$_$agg_func$_element_tuple(ARGS, M).");
+	
+	private static final ST NONBINDING_LITERAL_RESULT_RULE = Util.aspStringTemplate(
+			"$aggregate_result$($args$, $cmp_term$) :- $cmp_term$ $cmp_op$ AGG_VAL, $id$_$agg_func$_element_tuple($args$, AGG_VAL), $dependencies;separator=\", \"$.");
+	
+	private static final ST MAX_LITERAL_ENCODING = Util.aspStringTemplate(
+			MINMAX_ELEMENT_ORDERING +
+			"$id$_element_tuple_has_greater(ARGS, TPL) :- $id$_element_tuple_less_than(ARGS, TPL, _)." +
+			"$id$_max_element_tuple(ARGS, MAX) :- " +
+			"	$id$_element_tuple(ARGS, MAX), not $id$_element_tuple_has_greater(ARGS, MAX)."
+			);
+	
+	private static final ST MIN_LITERAL_ENCODING = Util.aspStringTemplate(
+			MINMAX_ELEMENT_ORDERING + 
+			"$id$_element_tuple_has_smaller(ARGS, TPL) :- $id$_element_tuple_less_than(ARGS, _, TPL)." +
+			"$id$_min_element_tuple(ARGS, MIN) :- " +
+			"	$id$_element_tuple(ARGS, MIN), not $id$_element_tuple_has_smaller(ARGS, MIN)."
+			);
+	//@formatter:on
+
+	private final ProgramParser parser = new ProgramParser();
+
+	public MinMaxEncoder(AggregateFunctionSymbol func) {
+		super(func, SetUtils.hashSet(ComparisonOperator.values()));
+		if (!(func == AggregateFunctionSymbol.MAX || func == AggregateFunctionSymbol.MIN)) {
+			throw new IllegalArgumentException("Encoder " + this.getClass().getSimpleName() + " can only encode min/max aggregates!");
+		}
+	}
+
+	@Override
+	protected InputProgram encodeAggregateResult(AggregateInfo aggregateToEncode, AggregateRewritingContext ctx) {
+		ST encodingTemplate = null;
+		if (this.getAggregateFunctionToEncode() == AggregateFunctionSymbol.MAX) {
+			encodingTemplate = new ST(MAX_LITERAL_ENCODING);
+		} else if (this.getAggregateFunctionToEncode() == AggregateFunctionSymbol.MIN) {
+			encodingTemplate = new ST(MIN_LITERAL_ENCODING);
+		} else {
+			// Note that this should definitely not happen due to the check in the constructor!
+			throw new UnsupportedOperationException("Cannot encode anything other than min/max aggregates!");
```

Comment:
tbh, I'm not really a fan of `oops` because of the type `RuntimeException`. In this case, if we hit the code, we clearly attempted an operation that is not supported, hence the `UnsupportedOperationException`. I guess it could be an `IllegalStateException` as well, but should be something a bit more specific than `RuntimeException`.